torch/distributed/checkpoint/state_dict.py

```diff
@@ -132,6 +131,7 @@ class _StateDictInfo(StateDictOptions):
     fsdp_modules: List[nn.Module] = field(default_factory=list)
 
 
+@functools.lru_cache(maxsize=None)
```

Comment:
would it be safer to set a maxsize here? otherwise this could lead to a (probably very small) memleak?

Reply:
When `maxsize` is None, the performance is faster as this means no LRU is used. `functools.cache` can achieve the same behavior but it is available only from Python 3.9. Theoretically, the get_state_dict should have the module every time so the cache of `_get_fqns` should stabilize after the first `get_state_dict`.

Reply:
I want to flag that this is not the case if `get_state_dict` is being used for different models in the same endpoint, and will likely be leaky.